handler/implementations/procSysKernel.go

```diff
@@ -492,6 +506,18 @@ func (h *ProcSysKernel) Write(
 
 	case "hostname":
 		return writeCntrData(h, n, req, nil)
+
+	case "shmall":
+		fallthrough
+	case "shmmax":
+		fallthrough
+	case "shmmni":
+		// The kernel only allows true root to write to /proc/sys/kernel/shm*.
+		// Root in the container's user-namespaces is not allowed to modify these
+		// values, even though they are namespaced via the IPC namespace.
+		// Therefore ask the passhthrough handler to enter all namespaces except
+		// the user-ns, as otherwise we get permission denied.
+		return h.Service.GetPassThroughHandler().WriteWithNS(n, req, domain.AllNSsButUser)
```

Comment:
Do we really need to push this value to kernel? Can't we emulate it instead as we do with other values avoiding writing the value to the kernel? I'm thinking about potential memory DoS attacks here...

Reply:
Yes, we want to push this value to the kernel because (1) it's namespaced via the IPC ns (so will only affect the container), and (2) it allows a user to restrict the amount of shared memory that a container can use. So in fact this helps mitigate DoS attacks.